### PR TITLE
Fixed signed int TIFF export in isis2std

### LINKED ISSUE #4897: isis2std bittype=s16bit exports a tiff with Type=UInt16
**ISIS version(s) affected**: 5.0.1

**Description**  
`isis2std bittype=s16bit` outputs UInt16 (16-bit unsigned integer) tif, not Int16 (signed) tif. 

**How to reproduce**  
Input: Int16 cube
```
$ gdalinfo h5851_0002_sr2.cal2.cub
Driver: ISIS3/USGS Astrogeology ISIS cube (Version 3)
Files: h5851_0002_sr2.cal2.cub
Size is 1008, 1018
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0, 1018.0)
Upper Right ( 1008.0,    0.0)
Lower Right ( 1008.0, 1018.0)
Center      (  504.0,  509.0)
Band 1 Block=1008x1018 Type=Int16, ColorInterp=Undefined
  NoData Value=-32768
  Mask Flags: 
  Metadata:
    BANDWIDTH=250.000000
    BANDWIDTH_UNIT=nm
    WAVELENGTH=600.000000
    WAVELENGTH_UNIT=nm
```
Running `isis2std bittype=s16bit` and its result
```
$ isis2std from=h5851_0002_sr2.cal2.cub to=h5851_0002_sr2.cal2.tif format=tiff bittype=s16bit
Group = Results
  OutputFileName = h5851_0002_sr2.cal2.tif
  InputMinimum   = 726.0
  InputMaximum   = 1602.0
End_Group

$ gdalinfo h5851_0002_sr2.cal2.tif
Driver: GTiff/GeoTIFF
Files: h5851_0002_sr2.cal2.tif
Size is 1008, 1018
Coordinate System is `'
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0, 1018.0)
Upper Right ( 1008.0,    0.0)
Lower Right ( 1008.0, 1018.0)
Center      (  504.0,  509.0)
Band 1 Block=1008x1 Type=UInt16, ColorInterp=Gray
```

**Additional context**  
Running `isis2std bittype=u16bit` results in a UInt16 tif, which looks fine.
```
$ isis2std from=h5851_0002_sr2.cal2.cub to=h5851_0002_sr2.cal2.tif format=tiff bittype=u16bit
Group = Results
  OutputFileName = h5851_0002_sr2.cal2.tif
  InputMinimum   = 726.0
  InputMaximum   = 1602.0
End_Group

$ gdalinfo h5851_0002_sr2.cal2.tif
Driver: GTiff/GeoTIFF
Files: h5851_0002_sr2.cal2.tif
Size is 1008, 1018
Coordinate System is `'
Image Structure Metadata:
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0, 1018.0)
Upper Right ( 1008.0,    0.0)
Lower Right ( 1008.0, 1018.0)
Center      (  504.0,  509.0)
Band 1 Block=1008x1 Type=UInt16, ColorInterp=Gray
```

The similar issue was reported in the past for both tiff and jpeg2000 formats: [#1735](https://github.com/USGS-Astrogeology/ISIS3/issues/1735)  
The signed 16 bit format is commented out in the source code:
https://github.com/USGS-Astrogeology/ISIS3/blob/7.0/isis/src/base/apps/isis2std/tsts/tiff/Makefile

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

Added the ability to write out the sample format to the TIFF header in TiffExporter.

## Related Issue

Fixes #4897

## Motivation and Context

If you attempt to write out a signed 16-bit integer tiff file with isis2st it will incorrectly write out an unsigned 16-bit integer tiff file. It will also write out the incorrect pixel values and any negative signed 16-bit values will under-flow to become large unsigned 16-but values.

## How Has This Been Tested?

We used isis2std to write out a signed16 bit tiff file, from a signed 16 but JunoCam image, and then used gdal translate to convert that back into an ISIS cube. The ISIS cube matched the previous ISIS cube, except it had been stretched to use the entire range of signed 16 bit values.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by AustinSanders — DISMISSED** [review]

**Comment by jessemapel** [comment]
We're working on a test conversion for this so that we can get test coverage on this as the original tests are commented out. Sorry, should have marked as draft.

File: isis/tests/FunctionalTestsIsis2Std.cpp [hunk 1]
```diff
@@ -0,0 +1,1054 @@
...
+
+  UserInterface options(APP_XML, args);
+  try {
+    isis2std(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to translate image: " << e.what() << std::endl;
+  }
+
+  // std2isis smashes the 16-bit output back down to 8-bit on re-ingest
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsIsis2Std.cpp:723]: See #1923 for more info about this issue.

File: isis/src/base/apps/isis2std/tsts/toobig/Makefile [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-commands:
-	if [ `$(APPNAME) from=$(INPUT)/large.lbl \
-	  to=$(OUTPUT)/isis2stdTruth1.png \
-	  mode=grayscale \
-	  format=png \
-	  stretch=linear \
-	   >& $(OUTPUT)/error.txt` ]; \
-	then \
-	  true; \
-	fi;
```
Comment (jessemapel) [inline isis/src/base/apps/isis2std/tsts/toobig/Makefile:14]: I don't think this is worth testing through the tests as it requires generating a file over 2 GB in size. So, this is the only test I didn't convert.

File: isis/src/base/objs/ImageExporter/ImageExporter.cpp [hunk 3]
```diff
@@ -97,15 +98,23 @@ namespace Isis {
    * @param outputName The filename of the output cube
    * @param quality The quality of the output from 0 to 100, defaults to 100
    * @param compression The compression algorithm used. Image format specific.
+   * @param ui The optional user interface to set the input image pixel range.
    */
   void ImageExporter::write(FileName outputName, int quality,
-                            QString compression) {
+                            QString compression, UserInterface *ui) {
     ProcessExport &p = process();
-    if (!p.HasInputRange()) p.SetInputRange();
```
Comment (jessemapel) [inline isis/src/base/objs/ImageExporter/ImageExporter.cpp:104]: Process::SetInputRange calls Application which caused seg-faults when running isis2std outside an application. Thankfully, this just calls Application::getUserInterface and then passes it to another function. so we have to add the new optional UserInterface argument and pass it on down through the different children.

File: isis/tests/FunctionalTestsIsis2Std.cpp [hunk 4]
```diff
@@ -0,0 +1,1054 @@
...
+
+  UserInterface options(APP_XML, args);
+  try {
+    isis2std(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to translate image: " << e.what() << std::endl;
+  }
+
+  QFile outputFile(outputBmpFilename + ".bmp");
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsIsis2Std.cpp:1052]: We could test this for all of the other extensions, but this is all the old tests did so I just went with this. This app already has plenty of tests.

File: isis/src/base/apps/isis2std/isis2std.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      }
+    }
+
+    FileName outputName = ui.GetFileName("TO");
+    int quality = ui.GetInteger("QUALITY");
+
+
+    QString compression;
+    if (format == "TIFF") {
+      compression = ui.GetString("COMPRESSION").toLower();
```
Comment (AustinSanders) [inline isis/src/base/apps/isis2std/isis2std.cpp:69]: Do we need to perform any validation on the compression input, or is that handled elsewhere?

File: isis/tests/FunctionalTestsIsis2Std.cpp [hunk 6]
```diff
@@ -0,0 +1,1054 @@
...
+#include "Portal.h"
+#include "SpecialPixel.h"
+#include "Statistics.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/isis2std.xml").expanded();
+static QString STD2ISIS_XML = FileName("$ISISROOT/bin/xml/std2isis.xml").expanded();
+
+class IsisTruthCube : public TempTestingFiles {
```
Comment (AustinSanders) [inline isis/tests/FunctionalTestsIsis2Std.cpp:25]: What's the rationale for defining the fixtures here instead of fixtures.cpp?

**Review by AustinSanders — DISMISSED** [review]
I think this is good to merge, just a few questions for my own understanding

File: isis/src/base/apps/isis2std/isis2std.cpp (hunk 5 above)
Comment (jessemapel) [reply isis/src/base/apps/isis2std/isis2std.cpp:69]: std2isis verifies the compressed output when it re-ingests the image.

File: isis/tests/FunctionalTestsIsis2Std.cpp (hunk 6 above)
Comment (jessemapel) [reply isis/tests/FunctionalTestsIsis2Std.cpp:25]: No real reason. It's faster to not add stuff to that as it causes re-compiling all of the tests whenever the header changes. If these could be useful elsewhere then I can move them.

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**